src/core/core.js

```diff
@@ -187,6 +187,7 @@ export default class Core {
 
         if (data[StorageKeys.DYNAMIC_FILTERS]) {
           this.globalStorage.set(StorageKeys.DYNAMIC_FILTERS, data[StorageKeys.DYNAMIC_FILTERS]);
+          this.globalStorage.set(StorageKeys.RESULTS_HEADER, data[StorageKeys.DYNAMIC_FILTERS]);
```

Comment:
I'm not sure I follow why we can't just configure ResultsSectionHeader to listen to DYNAMIC_FILTERS. Can you not turn off a specific callback listener with globalStorage.off? It looks like it takes a callback to turn off.

Reply:
It looks like, but the callback is completely ignored. On closer inspection though, what `globalStorage.off()` does is call `this._moduleDataContainer[moduleId].off(evt, cb);`, which is an instance of `ModuleData`. `ModuleData` doesn't override off(), and uses the implementation in its parent class EventEmitter. EventEmitter's off only takes a single argument, `off (evt) {`

Reply:
this is the source code for EventEmitter's off()
```
  off (evt) {
    delete this._listeners[evt];
    return this;
  }
```
each moduleId corresponds to one instance of `ModuleData`, so this ends up clearing every listener for a given moduleId

Reply:
during a given component's `unMount()` all of the component's children have their remove() called.

component.js
```
  unMount () {
    if (!this._container) {
      return this;
    }

    this._children.forEach(child => {
      child.unMount();
    });

    DOM.empty(this._container);
    this._children.forEach(c => c.remove());
    this._children = [];
    this.onUnMount();
  }
```

this `remove()` first recursively calls `remove()` on all of its children, and then removes itself using componentmanager's remove()

component.js
```
  remove () {
    this._children.forEach(c => c.remove());
    this.componentManager.remove(this);
  }
```